src/app/shared/services/geoNames.js

```diff
@@ -1,8 +1,12 @@
 export default class GeoNames {
-    constructor($http, $translate) {
+    constructor($http, $ngRedux) {
         this.$http = $http;
-        this.lang = $translate.use();
         this.baseURL = 'https://secure.geonames.org';
+        this.$ngRedux = $ngRedux;
+        $ngRedux.subscribe(() => {
```

Comment:
I don't think that subscribe to all redux changes is a good idea here. I think is better to get redux state before make any request, or maybe accept the language as argument